### ISSUE TITLE
Where should hypopg be installed?

### ISSUE THREAD

**Issue by pgiraud** [powa-team/powa-podman#19]
In the [SQL script](https://github.com/powa-team/powa-podman/blob/master/powa-archivist-git/install_all_powa_ext.sql) where extensions are installed, `hypopg` is installed in the `postgres` and in `template1`. Is there any reason for that?

In the powa-web interface, hypopg is marked as not installed.

![Screenshot from 2024-12-19 18-55-33](https://github.com/user-attachments/assets/0f0fa6e1-f793-45ce-83af-46b1beb80b14)

**Comment by rjuju** [comment]
> hypopg is installed in the postgres and in template1. Is there any reason for that?

yes, it's to ensure that hypopg is installed in a new database by default.  It's also in postgres in case people use this database rather than creating a new database.

> In the powa-web interface, hypopg is marked as not installed.

Yes, because I assumed that no one is going to use this image to try to optimize powa-archivist itself (something like https://github.com/powa-team/powa-archivist/commit/18d0b0c9670281f92d0d8f2dcad242f00a8887ce for instance).  And if someone wants to do having to install an extension is not going to be the hard part of the process.

**Comment by pgiraud** [comment]
> yes, it's to ensure that hypopg is installed in a new database by default. It's also in postgres in case people use this database rather than creating a new database.
> 

I understand. I don't know why by I got a case (with the `powa-dev-demo.yml` compose file) where I got a database where `hypopg` extension was not created. Weird.

> Yes, because I assumed that no one is going to use this image to try to optimize powa-archivist itself (something like [powa-team/powa-archivist@18d0b0c](https://github.com/powa-team/powa-archivist/commit/18d0b0c9670281f92d0d8f2dcad242f00a8887ce) for instance). And if someone wants to do having to install an extension is not going to be the hard part of the process.

As far as I understand, the table in configuration shows the extensions installed in the `powa` database.
https://github.com/powa-team/powa-web/blob/master/powa/config.py#L480

In 4.2.2, the extensions was marked as installed even if it was not installed in `powa` database.
![Screenshot from 2024-12-20 08-37-09](https://github.com/user-attachments/assets/0e462e2d-78ae-421a-b7ad-84534eda9944)
Screenshot obtained running powa-web 4.2.2 and powa-archivist-15:4.2.2 image.

**Comment by rjuju** [comment]
> I don't know why by I got a case (with the powa-dev-demo.yml compose file) where I got a database where hypopg extension was not created.

maybe the database was created before the hypopg part? if you have the full logs I could see if I see anything, otherwise you could look at the xmin of the various rows to try to infer what happened first.

> As far as I understand, the table in configuration shows the extensions installed in the powa database.
https://github.com/powa-team/powa-web/blob/master/powa/config.py#L480
>
> In 4.2.2, the extensions was marked as installed even if it was not installed in powa database.

yes I entirely overhauled the configuration page because it was quite misleading. there are now different sections based on the extension type. maybe a new section for "support extension" or something like that would be good, but we would need to connect to all databases to show the full information and that can be problematic. so maybe not showing the per database status, or showing it on another per database page.
